src/argsparser.c

```diff
@@ -301,6 +302,20 @@ void parse_recurring_delay(char *arg)
     free(x);
 }
 
+int isnumber(char *pin)
```

Comment:
> if string is numeric and of size 8, check checksum. if checksum mismatch, one-shot mode, else normal operation.

If checksum mismatch & "exhaustive" mode enabled, normal operation (it will bruteforce all 8 digits). Yep, this would require adding some more logics to "is_valid_pin" function.

> as for your comment

This is exactly what it will do + the comment above.

Maybe revert to the initial solution with new option? :)